types/src/data_types.rs

```diff
@@ -132,10 +137,13 @@ impl DataTypeNode {
             "MultiLineString" => Ok(Self::MultiLineString),
             "Polygon" => Ok(Self::Polygon),
             "MultiPolygon" => Ok(Self::MultiPolygon),
+            "Time" => Ok(Self::Time(None)),
 
             str if str.starts_with("Decimal") => parse_decimal(str),
             str if str.starts_with("DateTime64") => parse_datetime64(str),
             str if str.starts_with("DateTime") => parse_datetime(str),
+            str if str.starts_with("Time64") => parse_time64(str),
+            str if str.starts_with("Time") => parse_time(str),
```

Comment:
Looks like you could just return `Self::Time` here and remove `parse_time` fn, see https://clickhouse.com/docs/sql-reference/data-types/time